### PR TITLE
Add Cancel to IdenitfyUI, always call it

### PR DESCRIPTION
**Pull request by patrickxb**
r? @maxtaco 
cc @MarcoPolo

### CODE REVIEW

File: go/engine/identify.go [hunk 1]
```diff
@@ -94,6 +94,10 @@ func (e *Identify) SubConsumers() []libkb.UIConsumer {
 
 // Run starts the engine.
 func (e *Identify) Run(ctx *Context) error {
+
+	// always call Cancel to allow UI servers to cleanup.
+	defer ctx.IdentifyUI.Cancel()
```
Comment (maxtaco) [inline go/engine/identify.go]: what about `identify2_with_uid`?

**Comment by patrickxb** [comment]
@maxtaco PTAL

File: go/engine/buffered_identify_ui.go [hunk 2]
```diff
@@ -224,6 +225,14 @@ func (b *bufferedIdentifyUI) ReportTrackToken(t keybase1.TrackToken) error {
 	return b.raw.ReportTrackToken(t)
 }
 
+func (b *bufferedIdentifyUI) Cancel() error {
+	b.Lock()
+	defer b.Unlock()
+
+	// Cancel should always go through to UI server
+	return b.raw.Cancel()
```
Comment (maxtaco) [inline go/engine/buffered_identify_ui.go]: This won't wind up being called in `Identify2WithUID`.  I think that's fine but just double-checking with you.

**Comment by patrickxb** [comment]
Well, then Cancel() is only called on delegated identify uis.  I guess that's fine, but a little weird that the flow is different.

**Comment by maxtaco** [comment]
Yeah, that's a good point. OK, let's leave it.

**Comment by patrickxb** [comment]
I don't understand why Circle CI is failing since keybase1 protocol is vendored by kbfs.

According to @strib "the circle ci stuff does some craziness that pulls from both repos at once and they need to be in sync"

If that's true, I can't think of any way to get Circle CI to pass without breaking KBFS master.

@gabriel ?

**Comment by patrickxb** [comment]
@MarcoPolo gave me permission to break Circle CI and merge this.  

Seems like a monorepo is the best way to fix this going forward with mobile builds.

**Pull request merged**